framework/wazuh/core/tests/test_agent.py

```diff
@@ -585,18 +585,6 @@ def test_agent_reconnect_ko(socket_mock, send_mock, mock_queue):
         agent.reconnect(mock_queue)
 
 
-@patch('wazuh.core.agent.WazuhQueue')
-@patch('wazuh.core.wdb.WazuhDBConnection._send', side_effect=send_msg_to_wdb)
-@patch('socket.socket.connect')
-def test_agent_restart(socket_mock, send_mock, mock_queue):
-    """Test if method restart calls other methods with correct params."""
-    agent = Agent(0)
-    agent.restart()
-
-    # Assert methods are called with correct params
-    mock_queue.assert_called_once()
-
-
```

Comment:
Removing this test reduces the coverage:

![image](https://user-images.githubusercontent.com/37274979/121648983-5e3fa080-ca98-11eb-8cbc-f5e4e0b93fd0.png)

I understand what you are trying to do, but you could just merge both `test_agent_restart` and `test_agent_restart_ko` instead of just leaving `test_agent_restart_ko`.

Reply:
I have merged both and tested agent.restart with a MagicMock